dot-studio/svc_coord

```diff
@@ -51,42 +51,75 @@ function wait_for_svc_to_load() {
 }
 
 document "wait_for_port_to_listen" <<DOC
-  Wait for a port to be listening.
-
+  Wait for a port to be listening. If the port is not found in the set amount of time the function exits with 1
   @(arg:1) Port to wait for to be listening
-  @(arg:2) Wheter or not this process runs silently
+  @(arg:2) Number of seconds to wait to exit with 1. Default 60 seconds.
+  @(arg:3) Wheter or not this process runs silently
+
+  Example: Wait for a service to start listening on port 1234 for 1 minute
+  ---------------------------------------------
+  wait_or_fail_for_port_to_listen 1234 60
 DOC
 function wait_for_port_to_listen() {
   install_if_missing core/busybox-static netstat
+  SECOUNDS_WAITING=$2
+  if [ -z "$SECOUNDS_WAITING" ]
+  then
+    SECOUNDS_WAITING=60
+  fi
   
+  COUNTER=0
   while : ; do
     if netstat -an | grep "$1" | grep LISTEN >/dev/null 2>/dev/null
     then
       break
     else
       sleep 1
     fi
-    [[ "$2" != "silent" ]] && echo " => Waiting for port $1 to be listening"
+
+    if [ $COUNTER -ge $SECOUNDS_WAITING ]
+    then
+      echo " => Failed listening for port $1"
+      exit 1
+    fi
+    let COUNTER=COUNTER+1
+    [[ "$3" != "silent" ]] && echo " => Waiting for port $1 to be listening ($COUNTER of $SECOUNDS_WAITING)"
   done
 }
 
 document "wait_for_ok_response" <<DOC
-  Wait for the provided URL to response with status code (200)
+  Wait a set amount of time for the provided URL to response with status code (200). 
+  If the time expires then exit with 1
```

Comment:
You have to leave an empty line between the tittle and the description, just like Github commits. 😄